### ISSUE TITLE
Kong Req Transformation Bug Authorization header rename functionality

### ISSUE THREAD

**Issue by jeremyjpj0916** [Kong/kong#9418]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.1

### Current Behavior

When I use the request tranformer config.header.rename I can see its behavior flakes between including the bearer token from OAuth2 token vs transforming/renaming the specified header to take the Authorization Headers place:

Ex:
Not working ex(left kongs oauth2 token in the Authorization header):
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/31913027/189502472-0262758d-02e4-4628-8fbd-3823bf8b1df7.png">

When it works: (Note the Authorization header being reported with the Bearer XXXXXXXXXXXX replacement value.

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/31913027/189502506-29ae2d64-19bb-403f-a217-7b38982015ff.png">

This is being produced by having a backend "echo" service spit back seen http headers + body for me. 

### Expected Behavior

It should always replace the Authorization headers value to have: Bearer XXXXXXXXXXXX in it.

### Steps To Reproduce

```markdown
1. Make a Kong route + service proxy running OAuth2.0 CC + ACL + Req Transformation plugin on it.

2. Call the proxy with proper Authorization bearer header to succeed with OAuth2.0 Kong auth AND pass back a X-Venafi-Auth header with any kinda value in it.

3. Watch and see Kong be inconsistent with what the upstream receives with the transformation plugin in place with respect to the Authorization Header the upstream API receives.

Tranformer config ex:

{
  "service": null,
  "config": {
    "append": {
      "headers": [],
      "body": [],
      "querystring": []
    },
    "add": {
      "headers": [],
      "body": [],
      "querystring": []
    },
    "replace": {
      "uri": null,
      "headers": [],
      "body": [],
      "querystring": []
    },
    "rename": {
      "headers": [
        "X-Venafi-Auth:Authorization"
      ],
      "body": [],
      "querystring": []
    },
    "remove": {
      "headers": [],
      "body": [],
      "querystring": []
    },
    "http_method": null
  },
  "protocols": [
    "grpc",
    "grpcs",
    "http",
    "https"
  ],
  "enabled": true,
  "id": "edab2d9e-25aa-44fc-a49c-457322e35e94",
  "name": "request-transformer",
  "consumer": null,
  "created_at": 1662845032,
  "route": {
    "id": "3863f015-c7d7-49b7-b14e-1f776ebd5b49"
  },
  "tags": null
}
```
```


### Anything else?

_No response_

**Comment by jeremyjpj0916** [comment]
Looks like a fix is in the pipeline, goods stuff.

**Comment by jeremyjpj0916** [comment]
@windmgc  I tested your PR as a hot patch to our environment and can confirm it behaves as expected and fixed the issue.

**Comment by kikito** [comment]
Closing this issue then, thank you for reporting! 

The fix will only land on the 3.1 release though, unless we consider them critical enough to be backported.
